src/App.tsx

```diff
@@ -1,24 +1,245 @@
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    todoService.getTodos(USER_ID)
+      .then((todosFromServer) => {
+        setTodos(todosFromServer);
+      })
+      .catch(() => {
+        setErrorMessage(ErrorMessages.LOAD);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (errorMessage) {
+      const timeout = setTimeout(() => {
+        setErrorMessage(null);
+      }, 3000);
+
+      return () => clearTimeout(timeout);
+    }
+
+    return () => { };
+  }, [errorMessage]);
+
+  const handleAddTodo = useCallback(async (title: string) => {
+    setIsLoading(true);
+
+    try {
+      const newTodo = {
+        title,
+        userId: USER_ID,
+        completed: false,
+      };
+
+      setTempTodo({ ...newTodo, id: 0 });
+
+      const createdTodo = await todoService.createTodo(newTodo);
+
+      setTodos((currentTodos) => [...currentTodos, createdTodo]);
+    } catch {
+      setErrorMessage(ErrorMessages.ADD);
+    } finally {
+      setTempTodo(null);
+      setIsLoading(false);
+    }
+  }, []);
+
+  const handleRemoveTodo = useCallback(async (todoId: number) => {
+    setIsLoading(true);
+
+    try {
+      setTodos((currentTodos) => currentTodos
+        .map((todo) => (todo.id === todoId
+          ? { ...todo, isLoading: true } : todo)));
+
+      await todoService.deleteTodo(todoId);
+
+      setTodos((currentTodos) => currentTodos
+        .filter((todo) => todo.id !== todoId));
+    } catch {
+      setErrorMessage(ErrorMessages.DELETE);
+    } finally {
+      setIsLoading(false);
+    }
+  }, []);
+
+  const handleClearAllCompletedTodos = useCallback(async () => {
+    todos
+      .filter((todo) => todo.completed)
+      .map((todo) => handleRemoveTodo(todo.id));
+  }, [todos]);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!todoTitle.trim()) {
+      setErrorMessage(ErrorMessages.TITLE);
+
+      return;
+    }
+
+    handleAddTodo(todoTitle);
+    setTodoTitle('');
+  };
+
+  const handleToggleCompleted = useCallback(
+    async (todoId: number, completed: boolean) => {
+      setIsLoading(true);
+
+      try {
+        const updatedTodo = {
+          completed: !completed,
+        };
+
+        setTodos((currentTodos) => currentTodos
+          .map((todo) => (todo.id === todoId
+            ? { ...todo, completed: !completed, isLoading: true }
+            : todo
+          )));
+
+        await todoService.updateTodo(todoId, updatedTodo);
+
+        setTodos((currentTodos) => currentTodos
+          .map((todo) => (todo.id === todoId
+            ? { ...todo, isLoading: false } : todo)));
+      } catch {
+        setErrorMessage(ErrorMessages.UPDATE);
+      } finally {
+        setIsLoading(false);
+      }
+    },
+    [],
+  );
+
+  const handleToggleAllTodosCompleted = () => {
+    setIsLoading(true);
+
+    try {
+      todos.map((todo) => handleToggleCompleted(todo.id, todo.completed));
```

Comment:
Same issue as above. You are using .map method which returns an array, it would be better to use .forEach method because you don't need to return an array.